nixio/cmd/nixexplore.py

```diff
@@ -0,0 +1,622 @@
+                for s in secs:
+                    for prop in s.props:
+                        part = parts[1] if case_sensitive else parts[1].lower()
+                        pname = prop.name if case_sensitive else prop.name.lower()
+                        if part == pname if full_match else part in pname:
+                            print("[section: %s, type: %s, id: %s] >> " % (s.name, s.type, s.id), end="")
+                            prop.pprint()
+            else:
+                secs = find_section(f, p, case_sensitive)
+                if len(secs) == 0:
+                    props = find_props(f, p, case_sensitive)
+                    for sec in props.keys():
+                        for prop in props[sec]:
+                            print("[section: %s, type: %s, id: %s] >> " % (sec.name, sec.type, sec.id), end="")
+                            prop.pprint()
+                else:
+                    for s in secs:
+                        s.pprint(max_depth=arguments.depth)
+
+    f.close()
+    print("\n\n")
+
+
+def mdata_worker(arguments):
+    files = assemble_files(arguments)
+    for nf in files:
+        disp_metadata(nf, arguments)
+
+
+def find_data_entity(nix_file, arguments):
+    name_or_type = arguments.pattern if arguments.case_sensitive else arguments.pattern.lower()
+    classes = ["data_arrays"]  # , "multi_tags", "tags", "data_frames"]
+    entities = []
+    for b in nix_file.blocks:
+        for c in classes:
+            for e in getattr(b, c):
+                ename = e.name() if arguments.case_sensitive else e.name.lower()
+                etype = e.type() if arguments.case_sensitive else e.type.lower()
+                if arguments.full_match:
+                    if ename == name_or_type or etype == name_or_type:
+                        entities.append(e)
+                else:
+                    if name_or_type in ename or name_or_type in etype:
+                        entities.append(e)
+    return entities
+
+
+def get_ticks(dimension, extent):
+    ticks = None
+    dim_type = dimension.dimension_type
+    if dim_type == nix.DimensionType.Sample:
+        ticks = np.array(dimension.axis(extent))
+    elif dim_type == nix.DimensionType.Range:
+        ticks = np.array(dimension.ticks)
+    elif dim_type == nix.DimensionType.Set:
+        ticks = np.array(dimension.labels)
+        if len(ticks) == 0:
+            ticks = np.arange(0., extent, 1.0)
+    elif dim_type == nix.DimensionType.DataFrame:
+        ticks = np.array(dimension.ticks)
+    else:
+        print("Unsupported dimension type!")
+    return ticks
+
+
+def get_dim_label_and_unit(dimension):
+    dim_label = getattr(dimension, "label") if hasattr(dimension, "label") else ""
+    dim_unit = getattr(dimension, "unit") if hasattr(dimension, "unit") else ""
+    return dim_label, dim_unit
+
+
+def dump_oned(data, dimension, label, unit, format="%.6f", end="\n\n"):
+    assert(len(data.shape) == 2)
```

Comment:
Shouldn't this be `== 1` for oned?

More generally though, Assertions in scripts are a little user hostile.  They'll produce an AssertionError and traceback which isn't very helpful.  A friendlier error message should be printed if this happens.

Reply:
fair enough, avoided assertions